### DISCUSSION TITLE
SX1262 (DRJ1262T) Fail code -707

### DISCUSSION THREAD

**Discussion by NezhaZahri** [jgromes/RadioLib#541] [Q&A]
Hello,

I'm using the Lora SX1262 Module DRF1262T plugged into a Artemis Global Tracker which is a Redboard Sparkfun.
I'm trying to communicate to a module that has a SX1272 radio. I'm wondering if I'm supposed to match all the settings like code rate, sync word, preamble etc.. This is my first time working with radios.

I made sure the wiring were good, but I didn't plug the RST, BUSY,DIO1,DIO2 and SW pins of the SX1262, I'm wondering if it could be the reason i'm getting an error -707.
At the beginning I had a -2 error code because I inversed MISO and MOSI wires, but now I'm getting -707 error code.

Here's the code I wrote, please let me know if I forgot something: 

```c++
#include <SPI.h>
#include <RadioLib.h>
SX1262 radio = new Module(spiCS1,43,43,35); //(NSS,DIO1,NRST,BUSY)
int state = 0;

void setup(){
Serial.begin(115200);

  SPI.begin();
  
  // initialize SX1262 with default settings
  // carrier frequency:           915.0 MHz
  // bandwidth:                   125.0 kHz
  // spreading factor:            9
  // coding rate:                 7
  // sync word:                   0x12 (public network/LoRaWAN)
  // output power:                2 dBm
  // preamble length:             8 symbols
  state = radio.begin(915.0, 125.0, 9, 7, 0x12, 2, 8, 1.6 , false);
  if (state == RADIOLIB_ERR_NONE) {
    Serial.println(F("success!"));
  } else {
    Serial.print(F("failed, code "));
    Serial.println(state);
    while (true);
  }
}
```

**Comment by jgromes** [comment]
Two thoughts:

1. Don't call `SPI.begin()` in the Arduino sketch, the library does it internall. I don't know what happens on Redboard when you call SPI.begin twice.

2. Please run the sketch in [debug mode](https://github.com/jgromes/RadioLib/wiki/Debug-mode) with both debug and verbose enabled, and post the output. -707 is `RADIOLIB_ERR_SPI_CMD_FAILED`, so we will need to check what exactly went wrong.

To answer your question:

> I'm wondering if I'm supposed to match all the settings like code rate, sync word, preamble etc.

Yes, for LoRa you have to match the frequency, bandwidth, spreading factor, coding rate, sync word and preamble length.

**Reply by NezhaZahri** [reply]
This is what i'm getting in verbose and debug mode:
Again, i'm wondering if i'm supposed to plug the BUSY and RST pins maybe?
=~=~=~=~=~=~=~=~=~=~=~= PuTTY log 2022.06.29 15:27:30 =~=~=~=~=~=~=~=~=~=~=~=
M	SX126x
CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

CMD	80	
DATW	0	AA	

failed, code -707
^A  DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

CMD     80
DATW    0       AA

failed, code -707

**Reply by jgromes** [reply]
> Again, i'm wondering if i'm supposed to plug the BUSY and RST pins maybe?

Oh, I missed that in your original post. Yes, you absolutely have to connect the BUSY pin. NRST is optional, but if you're not using it, put the `RADIOLIB_NC` macro at the appropriate position in the Module constructor. You also have to connect the DIO1 pin as well, because that tells the Redboard when a packet is received, transmission is finished, etc. Finally, if I recall correctly, the SW pin control the RF switch, so you will have to switch it on and off to transmit and receive.

**Reply by NezhaZahri** [reply]
I soldered all the pins accordingly, it works well! 
Thanks! 👍

**Reply by NezhaZahri** [reply]
Btw you are talking about the SW pin, but it seems that it is connected internally, I should leave DIO2 open right?
And actually not use this function: 
setDio2AsRfSwitch()

<img width="591" alt="image" src="https://user-images.githubusercontent.com/101749432/176735692-a9ab134c-efd1-48b8-9688-1b778e3d0775.png">

**Reply by jgromes** [reply]
I googled the schematic, here's the relevant part:
![Screenshot_26](https://user-images.githubusercontent.com/3407192/176747864-b6840aec-442f-4321-8c20-e34fb1f84fa1.png)

You have to pull the SW pin to 3.3V in order for the RF switch to work.

**Comment by NezhaZahri** [comment]
So I've been testing the transmission and reception and it seems to me quite randomly unreliable.
Sometimes it works well, sometime I'm getting timeouts, sometime error -1.  (The most often)

![image](https://user-images.githubusercontent.com/101749432/177393539-94be8080-dd45-4646-b269-849c4c68a685.png)


I was wondering if there is anything to make the code more robust.

![image](https://user-images.githubusercontent.com/101749432/177393191-640833ab-5cc3-49cc-a05a-c82603343425.png)

Let me know if i should create a new discussion thread.

**Reply by jgromes** [reply]
Random errors usually points to hardware issues - bad soldering, or unstable/insufficient power source. This is especially true for error -1 (unknown error), which marks all states that should be impossible to get to with properly working hardware.
